Extensions/Lighting/lightruntimeobject-pixi-renderer.js

```diff
@@ -94,4 +115,77 @@ gdjs.LightRuntimeObjectPixiRenderer.prototype.updateVertexBuffer = function () {
         this._object.y,
       ])
     );
+//  }
+};
+
+gdjs.LightRuntimeObjectPixiRenderer.prototype.raycastTest = function () {
+  var result = [];
+  var center = [this._object.x + this._radius, this._object.y + this._radius];
+  var halfOfDiag = Math.sqrt(2) * this._radius;
+
+  this._manager.getAllObstaclesAround(this._object, this._radius, result);
+  var hitBoxes = result.map(function (item) {
+    return item.owner.hitBoxes;
+  });
+  hitBoxes.push(this._object.getHitBoxes());
+
+  var polygons = hitBoxes.flat(1);
+  var vertices = polygons.map(function (poly) {
+    return poly.vertices;
+  }).flat(1);
+
+  function _calculatePOI(angle) {
+    var minDist = halfOfDiag;
+    var minPOI = [];
+    for (var poly of polygons) {
+      var poi = gdjs.Polygon.raycastTest(
+        poly,
+        center[0],
+        center[1],
+        center[0] + halfOfDiag * Math.cos(angle),
+        center[1] + halfOfDiag * Math.sin(angle)
+      )
+
+      if (poi.closeSqDist <= minDist) {
+        minDist = poi.closeSqDist;
+        minPOI.length = 0;
+        minPOI.push.apply(minPOI, [poi.closeX, poi.closeY]);
+      } 
+    }
+    console.log(minPOI);
+    return minPOI;
+  }
+
+  var _raycastResult = [];
+
+  for (var vertex of vertices) {
+    var xdiff = vertex[0] - center[0];
+    var ydiff = vertex[1] - center[1];
+    var angle = Math.atan(ydiff/xdiff);
+
+    if (xdiff < 0) angle += Math.PI;
+    if (xdiff >= 0 && ydiff < 0) angle += 2 * Math.PI;
+
+    _raycastResult.push({
+      vertex: _calculatePOI(angle),
+      angle: angle
+    });
+
+    // TODO: Check whether we need to raycast these two extra rays or not.
+    _raycastResult.push({
+      vertex: _calculatePOI(angle + 0.01),
+      angle: angle + 0.01
+    });
+    _raycastResult.push({
+      vertex: _calculatePOI(angle - 0.01),
+      angle: angle - 0.01
+    });
+  }
+
+  return _raycastResult.sort(function (a, b) {
+    if (a.angle < b.angle) return -1;
+    if (a.angle > b.angle) return 1;
```

Comment:
Your function is not returning anything if the angle are equal. Return 0 in this case.